ioeffect-cats/src/main/scala/scalaz/ioeffect/catz.scala

```
@@ -0,0 +1,64 @@
+package scalaz.ioeffect
+
+import cats.effect
+import cats.effect.Effect
+import cats.syntax.all._
+
+import scala.util.control.NonFatal
+
+object catz extends RTS {
+
+  implicit val catsEffectInstance = new Effect[Task] {
+    def runAsync[A](
+      fa: Task[A]
+    )(cb: Either[Throwable, A] => effect.IO[Unit]): effect.IO[Unit] =
+      effect.IO {
+        unsafePerformIO(
+          fa.attempt[Throwable].flatMap { a =>
+            IO.async[Throwable, Unit] { r =>
+              cb(a.toEither).unsafeRunAsync {
+                case Right(r2) => r(ExitResult.Completed(r2))
+                case Left(l2)  => r(ExitResult.Failed(l2))
+              }
+            }
+          }
+        )
+      }.attempt.void
```

What does `attempt` get us here?

---

Swallows errors. `.runAsync` requires this.